src/core/thread/mle_router.cpp

```diff
@@ -423,17 +423,14 @@ ThreadError MleRouter::ResetAdvertiseInterval(void)
 {
     uint32_t advertiseDelay;
 
-    VerifyOrExit(mAdvertiseInterval != kAdvertiseIntervalMin || !mAdvertiseTimer.IsRunning(), ;);
```

Comment:
Removing this check means that the code below will always be executed for each call to `ResetAdvertiseInterval`.  There is no need to reset the Trickle timer when `mAdvertiseInterval` is already set to the minimum.  Also, without the check, successive calls to `ResetAdvertiseInterval` can cause the timer to never fire (since `mAdvertiseTimer` keeps getting started with a new delay).